### PR TITLE
Update Settings Scaffold

### LINKED ISSUE #1437: Update Settings Scaffold
Recent versions of Django have significantly cut down the number of default settings required to get a site up and running.  We should copy that to keep the framework as close to Django as possible.

Further to that, splitting up the settings into four distinct sections should further aid readability:
1. Django
1. Third Party
1. Opal
1. Project

### PR DESCRIPTION
**Pull request by ghickman**
Base the Django settings scaffold file off Django 1.10's template since it's pinned in requirements.txt.  Split the settings up into sections.

Fixes #1437

### CODE REVIEW

File: opal/scaffolding/scaffold/app/settings.py.jinja2 [hunk 1]
```diff
@@ -150,108 +78,154 @@ TEMPLATES = [
...
+            'HOST': '',
+            'PORT': ''
         }
+    }
+
+# Password validation
+# https://docs.djangoproject.com/en/{{ docs_version }}/ref/settings/#auth-password-validators
+AUTH_PASSWORD_VALIDATORS = [
+    {
+        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
```
Comment (fredkingham) [inline opal/scaffolding/scaffold/app/settings.py.jinja2]: @ghickman am I right in thinking we'll have to change https://github.com/openhealthcare/opal/blob/84b809a4a29b84e97730018de9e49517f2564ef8/opal/scaffolding/scaffold/app/management/commands/createopalsuperuser.py.jinja2 for this?

(do validators trigger only via the form, is it a db thing?)

File: opal/scaffolding/scaffold/app/settings.py.jinja2 [hunk 2]
```diff
@@ -1,116 +1,49 @@
...
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 PROJECT_PATH = os.path.realpath(os.path.dirname(__file__))
 
-DEBUG = True
-
-ADMINS = (
-    # ('Your Name', 'your_email@example.com'),
-)
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/{{ docs_version }}/howto/deployment/checklist/
```
Comment (davidmiller) [inline opal/scaffolding/scaffold/app/settings.py.jinja2]: I like `{{ docs_version }}` as an idea, but are we definitely passing this in ?

File: opal/scaffolding/scaffold/app/settings.py.jinja2 [hunk 3]
```diff
@@ -150,108 +78,154 @@ TEMPLATES = [
...
 OPAL_LOG_OUT_MINUTES = 15
 OPAL_LOG_OUT_DURATION = OPAL_LOG_OUT_MINUTES*60*1000
 
-# Begins OPAL optional settings
 # OPAL_EXTRA_HEADER = ''
 # OPAL_EXTRA_APPLICATION = ''
 
-# Uncomment this to swap out the logo used by this application
+# Uncomment this if you want to swap the logo used by this application.
 # OPAL_LOGO_PATH = 'img/ohc-trans.png'
```
Comment (davidmiller) [inline opal/scaffolding/scaffold/app/settings.py.jinja2]: Maybe we should link to our docs here ?

Comment (davidmiller) [reply opal/scaffolding/scaffold/app/settings.py.jinja2]: http://opal.openhealthcare.org.uk/docs/v0.10.0/reference/settings/#opal_logo_path

**Pull request merged**